path_login.go

```diff
@@ -45,14 +45,58 @@ func pathLogin(b *backend) *framework.Path {
 			},
 		},
 		Callbacks: map[logical.Operation]framework.OperationFunc{
-			logical.UpdateOperation: b.pathLoginUpdate,
+			logical.UpdateOperation:      b.pathLoginUpdate,
+			logical.ResolveRoleOperation: b.pathResolveRole,
+		},
+
+		HelpSynopsis:    pathLoginRoleSyn,
+		HelpDescription: pathLoginRoleDesc,
+	}
+}
+
+func pathLogin(b *backend) *framework.Path {
```

Comment:
nit: do we need two path functions for this? I believe we can use a slice like `[]*framework.Path{..}` to add more paths.